whylog/config/__init__.py

```diff
@@ -149,21 +181,27 @@ def get_parsers_for_log_type(self, log_type):
 
 
 class InvestigationStep(object):
-    """
-    Represents rules, parsers and locations of logs which are necessary
-    to find and parse log files with potential causes.
-    """
-
-    def __init__(self, parsers, rules, log_location, effect_time):
-        pass
-
-    def get_clues(self, line):
+    def __init__(self, concatenated_parser, effect_time, earliest_cause_time):
+        self._concatenated_parser = concatenated_parser
```

Comment:
should be a more abstract variable, like "searcher" or something. I think `ConcatenatedParser` is a name of the class, not interface.